impeller/typographer/glyph_atlas.h

```diff
@@ -142,9 +142,22 @@ class GlyphAtlas {
   ///
   FontGlyphAtlas* GetOrCreateFontGlyphAtlas(const ScaledFont& scaled_font);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Retrieve the generation id for this glyph atlas.
+  ///
+  ///             The generation id is used to match with a TextFrame to
+  ///             determine if the frame is guaranteed to already be populated
+  ///             in the atlas.
+  size_t GetAtlasGeneration() const;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Update the atlas generation.
+  void SetAtlasGeneration(size_t value);
+
  private:
   const Type type_;
   std::shared_ptr<Texture> texture_;
+  size_t generation_ = 0;
```

Comment:
Do you think we should use std::optional here to signify atlases that have not had their generations set?  Or maybe put the generation in the constructor?

Reply:
an atlas always have a generation though.

Reply:
> an atlas always have a generation though.

It does, but an atlas that has been created but hasn't had `SetAtlasGeneration` called has an incorrect, but valid, generation. I'm suggesting that we make having an invalid generation explicit, or removing the possibility of having an invalid generation by lifting it to the constructor.

Reply:
Done